### PR TITLE
VASP update state change refactor

### PR DESCRIPTION
**Pull request by pdeziel**
This adds a new function `UpdateVerificationStatus` which updates the verification state of a VASP and appends an entry to the audit log to record the state change. State changes have been refactored to use this new function instead of just assigning a new value to `VASP.VerificationStatus`.

### CODE REVIEW

File: pkg/gds/admin.go
Comment (bbengfort) [inline pkg/gds/admin.go:564]: I'm working on the authentication middleware right now that will add the user email to the request context. Would you prefer to hold off on finishing this PR until that's merged and then rebase with it to implement it, or create a story to do this later and merge this PR without waiting for the middleware?

File: pkg/gds/gds.go
Comment (bbengfort) [inline pkg/gds/gds.go:116]: This is also the zero value so this entire line could be safely omitted, but I like that you've added it here explicitly for clarity.

File: pkg/gds/models/v1/models.go
Comment (bbengfort) [inline pkg/gds/models/v1/models.go:65]: I have a slight preference that the verification status gets changed before the audit log is appended, just in case we typo the previous state guard in `AppendAuditLog` but it's not a big deal.

**Review by bbengfort — APPROVED** [review]
Well done! Thank you for so thoroughly searching through the code to find all the VASP state changes! I really liked the descriptions you added and you were spot on with "email" vs "automated" as the source. 

There is one place that I would also like to add an audit trail; but you wouldn't have been able to find it easily since it doesn't actually change the VASP state. I'd like to track when emails are "resent" to the users:

https://github.com/trisacrypto/directory/blob/main/pkg/gds/admin.go#L901

My thought is that we would add a different audit message to each one of the cases in the `switch` for the email to send. The previous and current states will be the same. 

Note that many of the state changes imply emails being sent; however if it's easier, we could add auditing to the email manager so we know exactly what emails were sent. If this is your preference, then let's create a story for that and do it in a different PR.

File: pkg/gds/admin.go
Comment (pdeziel) [reply pkg/gds/admin.go:564]: Seems like a simple enough change so I'll hold off on merging this PR until the middleware one is done.

**Comment by pdeziel** [comment]
@bbengfort Thanks for getting to the review so quickly! I'll hold onto this PR for now until I can integrate the user email. As for your other question, I'm leaning towards creating another story for the email audit log and handling that in another PR.

**Comment by pdeziel** [comment]
@bbengfort I've rebased and added the email from the context, mind taking a quick look to verify that the rebase is correct and the claims error handling is ok?

**Review by bbengfort — APPROVED** [review]
Thank you for holding off until you had access to the user claims from the context! I rebased this with main since there was a merge conflict; would you mind taking a look to make sure nothing got dropped? Feel free to merge when you're ready.

**Pull request merged**
